tests/utils.rs

```diff
@@ -0,0 +1,33 @@
+#![cfg_attr(
+    feature = "dev",
+    allow(dead_code, unused_variables, unused_imports, unreachable_code)
+)]
+#![deny(clippy::all)]
+
+use check_protocols::{run_check_protocols, Context, ExitCode, R};
+use pretty_assertions::assert_eq;
+use std::fs;
+use test_utils::{trim_margin, TempFile};
+
+pub fn test_run_check_protocols(script: &TempFile, protocol: &str) -> R<(ExitCode, String)> {
```

Comment:
this function is only used 3 times in `tests/run.rs`, could it be made private in favor of only calling `test_run` there? i'm mostly wondering because i think the naming between the two functions isn't very clear.

it looks like you're using it so that you can get access to the script path since it appears in the error message you assert on, so maybe this could be called `test_run_with_tempfile`?

Reply:
Good point. Comparing with `test_run` it also doesn't call `compare_results`. I'll rename it to what you said.